kratos/python_scripts/python_solver.py

```diff
@@ -54,6 +54,13 @@ def AddVariables(self):
         """
         pass
 
+    def AddAdditionalHistoricalVariables(self, list_additional_variables):
```

Comment:
I completely forgot, this fct cannot return a list as a solver might have multiple ModelParts!
Instead it should be a map with name-mp to name-var

thx to pybind this should also work from c++

Reply:
I agree with you in principle, though there will be not so much gain in general, thats why I did not do it at the beginning. At least in FluidsDynApp and StructuralApp this will always be the main model part. Maybe there are other solvers that could benefit from that...

Reply:
I think this was a blocker last time
consider that there are many solvers already that use multiple `ModelParts`: DEM, MPM, FSI are the ones I know about

and changing an interface in the core is usually a huge pain

Reply:
Remember that: https://github.com/KratosMultiphysics/Kratos/pull/9121

Reply:
> Remember that: #9121

how is it related?

Reply:
In adding variables needed

Reply:
I see that #9121 it can be used inside this function, but this PR (#9117) is about the interface, so one level higher

Reply:
Ok @philbucher. Then the problem is how the process knows in which model part the variable should be added... I dont see how the process could consistently know this except if this is given as a variable in the process itself which kind of defeats the purpose of the PR... maybe we have to stick to something like https://github.com/KratosMultiphysics/Kratos/blob/71fd000125346170b782ed4ef1fd8f4cc2834660/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py#L134 but adding the option to specify the model part

Reply:
hm wait, which process doesn't know which ModelPart it is using? Afaik all the processes that use a ModelPart have a filed `model_part_name` in the input

Reply:
hmm yes, but I dont know if in all cases model part will already exist at that point. For sure not if the model part name is a submodelpart, but then maybe we can try to add it to the parent model part as they share the variables list... That might work.

Reply:
I would say that the solver needs to ensure that the ModelParts are existing at this point
I think most of the time we do this already

Yes adding to a parent or root ModelPart should be sufficient

Reply:
added a version with the map, what do you think?